app/src/main/java/org/android/go/sopt/di/RepositoryModule.kt

```diff
@@ -0,0 +1,16 @@
+package org.android.go.sopt.di
+
+import dagger.Binds
+import dagger.Module
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import org.android.go.sopt.data.repository.UserRepositoryImpl
+import org.android.go.sopt.domain.repository.UserRepository
+
+@Module
+@InstallIn(SingletonComponent::class)
+interface RepositoryModule {
+
+    @Binds
+    fun provideRepositoryImpl(repositoryImpl: UserRepositoryImpl): UserRepository
```

Comment:
이렇게 리턴을 하게 되면 [어플리케이션 스콥에서 싱글턴을 보장할 수 없습니다](https://developer.android.com/training/dependency-injection/hilt-android#component-scopes). 혹시 의도한 것이라면 어떤 이유에서 이와 같이 설계를 했는 지 알 수 있을까요? 

```suggestion
    @Binds
    @Singleton
    fun provideRepositoryImpl(repositoryImpl: UserRepositoryImpl): UserRepository
```

Reply:
UserRepositoryImpl이 클래스 인스턴스임인 간과한 코드입니다. 해당부분 수정해놓도록 하겠습니다! 

++ 또한 이름도 잘못되었네요 provide대신 bind가 적절한 네이밍일 것 같습니다.